frontend/src/helpers/dataOverlay.js

```diff
@@ -0,0 +1,97 @@
+const DEFAULT_GENE_COLOR = '#feb';
+const DEFAULT_METABOLITE_COLOR = '#9df';
+
+const DATA_TYPES_COMPONENTS = {
+  transcriptomics: {
+    componentType: 'gene',
+    className: 'enz',
+    defaultColor: DEFAULT_GENE_COLOR,
+  },
+  metabolomics: {
+    componentType: 'metabolite',
+    className: 'met',
+    defaultColor: DEFAULT_METABOLITE_COLOR,
+  },
+};
+
+/*
+ * @param { Blob } file
+ *
+ * @example
+ * try {
+ *   const data = await parseCustomFile(file);
+ * } catch (e) {
+ *   console.error(e)
+ * }
+ */
+const parseFile = async (file) => {
+  let tissues = [];
+  const levels = { 'n/a': 'n/a' };
+
+  const text = await file.text();
+
+  const lines = text.split(/\r?\n/);
+  let indexLine = 1;
+  // fetch Tissue
+  if (lines[0].split('\t').length !== 1) {
+    const arrLine = lines[0].split('\t');
+    const v = Number(arrLine[1]);
+    if (Number.isNaN(v)) {
+      tissues = lines[0].split('\t');
+      tissues.shift();
+      lines.shift();
+    } else {
```

Comment:
Would it be possible to have this type of check in the data validation stage?

Reply:
All validation will be moved into https://github.com/MetabolicAtlas/MetabolicAtlas/blob/35878b1f76a52469954cf891d64acb535c17d45a/frontend/src/components/explorer/mapViewer/DataOverlayValidation.vue